extensions/iceberg/s3/src/main/java/io/deephaven/iceberg/util/AsyncHttpClientProperties.java

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import java.time.Duration;
+import java.util.Map;
+import org.apache.iceberg.util.PropertyUtil;
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.services.s3.S3AsyncClientBuilder;
+
+class AsyncHttpClientProperties {
+
+    // Add option for CRT and inject this property
+    public static final String CLIENT_TYPE = "http-client.async.client-type";
+    public static final String CLIENT_TYPE_CRT = "aws-crt";
+    public static final String CLIENT_TYPE_NIO = "netty-nio";
+    public static final String CLIENT_TYPE_DEFAULT = CLIENT_TYPE_NIO;
+
+    public static final String READ_TIMEOUT_MS = "http-client.async.read-timeout-ms";
+    public static final String WRITE_TIMEOUT_MS = "http-client.async.write-timeout-ms";
+    public static final String CONNECTION_TIMEOUT_MS = "http-client.async.connection-timeout-ms";
+    public static final String MAX_CONCURRENCY = "http-client.async.max-concurrency";
```

Comment:
I wonder if we should prefer to follow the AWS pattern where the client type is part of the key:

```
http-client-async.netty.read-timeout-ms
```
v